### DISCUSSION TITLE
🌏 Linguistically typed languages

### DISCUSSION THREAD

**Discussion by hejny** [webgptorg/promptbook#53] [Concepts]
There are [no types in Promptbook](https://github.com/webgptorg/promptbook/discussions/52), but we can do this thing quite differently from their point of view. Make a type system from a linguistic point of view.

- Instead of primitives booleans, numbers, strings, symbols,... should be morphemes and lexemes.
- Instead of objects, words, nouns, verbs, adjectives,...
- Instead of complex structures, complex sentence types and structures or literary types like poem, essey,...

But maybe this is a stupid or unworkable idea?

## Questions

- How to deal with multiple human languages
- How to check this
- How to do retyping
- What about infinitum

## PIP 0002

See [promptbook improvement proposal PIP-0002](https://github.com/webgptorg/book/tree/main/pips/pip-0002) how to implement this in Book language.

**Comment by hejny** [comment]
## 🧪 What we tried, who we asked, is this concept a thing somewhere?


*(Describe here)*

**Reply by cie** [reply]
https://en.wikipedia.org/wiki/Attempto_Controlled_English

**Comment by hejny** [comment]
[🛴]

**Comment by hejny** [comment]
https://github.com/webgptorg/book/blob/main/pips/pip-0002.md
